src/atoms/Checkbox/Checkbox.web.js

```diff
@@ -1,3 +1,275 @@
+      return 'shade.950';
+    },
+    size({ size }) {
+      switch (size) {
+        case 'large':
+          return 'xlarge';
+        case 'medium':
+          return 'large';
+        case 'small':
+          return 'medium';
+        default:
+          return 'medium';
+      }
+    },
+  },
+  descriptionText: {
+    margin(size) {
+      switch (size) {
+        case 'large':
+          return [0.5, 0, 0, 4];
+        case 'medium':
+          return [0.5, 0, 0, 3.5];
+        case 'small':
+          return [0, 0, 0, 3];
+        case 'xsmall':
+          return [0, 0, 0, 2.5];
+        default:
+          return [0, 0, 0, 3.5];
+      }
+    },
+    size(checkboxSize) {
+      switch (checkboxSize) {
+        case 'large':
+          return 'medium';
+        case 'medium':
+          return 'xsmall';
+        default:
+          return 'small';
+      }
+    },
+    color({ disabled, errorText }) {
+      if (errorText) {
+        return 'negative.900';
+      }
+      if (disabled) {
+        return 'shade.930';
+      }
+      return 'shade.950';
+    },
+  },
+  backdrop: {
+    backgroundColor({ theme, state, isChecked, disabled, variantColor }) {
+      if (disabled) {
+        return 'transparent';
+      }
+
+      switch (state) {
+        case 'hover':
+          if (!isChecked) {
+            return theme.colors.tone[930];
+          }
+          return getColor(theme, `${variantColor}.920`);
+        case 'focus':
+          if (!isChecked) {
+            return theme.colors.tone[940];
+          }
+          return getColor(theme, `${variantColor}.930`);
+        case 'active':
+          if (!isChecked) {
+            return theme.colors.tone[940];
+          }
+          return getColor(theme, `${variantColor}.940`);
+        default:
+          return 'transparent';
+      }
+    },
+    width: {
+      large: makePxValue(3.5),
+      medium: makePxValue(3),
+      small: makePxValue(2.5),
+      xsmall: makePxValue(2),
+    },
+    height: {
+      large: makePxValue(3.5),
+      medium: makePxValue(3),
+      small: makePxValue(2.5),
+      xsmall: makePxValue(2),
+    },
+    borderRadius: {
+      large: makePxValue(1.75),
+      medium: makePxValue(1.5),
+      small: makePxValue(1.25),
+      xsmall: makePxValue(1),
+    },
+  },
+  title: {
+    color({ disabled }) {
+      if (disabled) {
+        return 'shade.950';
+      }
+      return 'shade.980';
+    },
+  },
+};
+
+const Checkbox = ({
+  defaultChecked,
+  checked: externalChecked,
+  onChange,
+  disabled,
+  title,
+  size,
+  variantColor,
+  helpText,
+  errorText,
+  testID,
+  name,
+  id,
+}) => {
+  if (isDefined(defaultChecked) && isDefined(externalChecked)) {
+    throw Error('One of defaultChecked or checked should be supplied.');
+  }
+
+  const [isChecked, setIsChecked] = useState(defaultChecked || false);
+
+  const onCheckChange = useCallback(() => {
+    if (isDefined(externalChecked)) {
+      onChange(!externalChecked);
+      return;
```

Comment:
Why existing early when `externalChecked` is defined?

Reply:
for not managing state in the component as `externalChecked` is meant for un-controlled comp.